core/services/workflows/models.go

```diff
@@ -200,16 +233,68 @@ func Parse(yamlWorkflow string) (*workflow, error) {
 		}
 	}
 
-	triggerSteps := []*triggerCapability{}
+	triggerSteps := []*StepDefinition{}
 	for _, t := range spec.Triggers {
-		triggerSteps = append(triggerSteps, &triggerCapability{
-			stepDefinition: t,
-		})
+		tt := t
+		triggerSteps = append(triggerSteps, &tt)
 	}
-	wf := &workflow{
-		spec:     &spec,
+	wf := &StaticWorkflow{
+		Spec:     &spec,
 		Graph:    g,
-		triggers: triggerSteps,
+		Triggers: triggerSteps,
 	}
 	return wf, err
 }
+
+// createExecutableWorkflow converts a StaticWorkflow to an executable workflow
+// by adding metadata to the vertices that is owned by the workflow runtime.
+func createExecutableWorkflow(wf2 *StaticWorkflow) (*workflow, error) {
```

Comment:
I would still consider this static evaluation, it's still a pure function. I think it'd be good to have this within `chainlink-common` too.

Reply:
The big difference between this function and `ParseStaticWorkflow` is that one  ensures the workflow matches our schema, the other ensures that the workflow is a valid graph (in a workflow sense)

Reply:
the workflow is mutated by the engine. i'm opening to naming suggestions. i don't want to change `workflow` though because one of the goals is minimal surgery to decouple the mutable state from the static state.

Reply:
re: moving this to common. i don't want to do that do because the mutable state is really tightly coupled to the engine. We'd have to prompt all the private fields to public, and it don't really make sense to separate it from the engine impl. (i start down that road before discovering that its possible to decouple the static and mutable state via an intermediate representation.)